Revise Swift entries (#929)
Codecademy/docs: content/swift/concepts/operators/terms/advanced-operators/advanced-operators.md

  - 'paths/build-ios-apps-with-swiftui'
---

> **Advanced operators** in Swift provide the programmer with increased functionality and complex value manipulation. They build upon the concepts covered in the basic operators wiki and include compound assignments, bitwise, and custom operators, to name a few.

Reviewer: ```suggestion
**Advanced operators** in Swift provide the programmer with increased functionality and complex value manipulation. They build upon the [basic operators](https://www.codecademy.com/resources/docs/swift/operators/basic-operators) and include compound assignments, bitwise, and custom operators.
```

## Compound Assignment Operator

> Compound assignment operators combine the assignment operator `=` with another separate operation, for example, `+`. The following operations are allowed `+`, `-`, `/`, `*`, `%`,

Reviewer: ```suggestion
Compound assignment operators combine the assignment operator `=` with another separate operation, for example, `+`. The following operations are allowed:
```

let result = firstBinary | secondBinary // 01111101
```

### XOR Operator

> Bitwise XOR `^` operator, also known as 'exclusive OR', compares two binary numbers. It will return a new number with each bit set to `1` if the bits of the original numbers are different, or exclusive. `0`'s will be placed anywhere that the two bits are equivalent.

Reviewer: ```suggestion
Bitwise XOR `^` operator, also known as 'exclusive OR', compares two binary numbers. It will return a new number with each bit set to `1` if the bits of the original numbers are different, or exclusive. `0`s will be placed anywhere that the two bits are equivalent.
```

// Output: "These are different students"
```

The `==` operator is used to logically compare `studentId` properties of two students.

> Swift also allows us to override most of the common operators. For a complete list, visit the [Swift Reference Manual](https://docs.swift.org/swift-book/ReferenceManual/LexicalStructure.html#ID418).

Reviewer: ```suggestion
Swift also allows us to override most of the common operators. A complete list of them can be found in the [Swift Reference Manual](https://docs.swift.org/swift-book/ReferenceManual/LexicalStructure.html#ID418).
```
